driver-core/src/main/com/mongodb/internal/connection/PowerOfTwoBufferPool.java

```diff
@@ -51,27 +110,37 @@ public PowerOfTwoBufferPool() {
      * @param highestPowerOfTwo the highest power of two buffer size that will be pooled
      */
     public PowerOfTwoBufferPool(final int highestPowerOfTwo) {
+        this(highestPowerOfTwo, DEFAULT_MAX_IDLE_TIME_MINUTES, TimeUnit.MINUTES);
+    }
+
+    /**
+     * Construct an instance.
+     *
+     * @param highestPowerOfTwo the highest power of two buffer size that will be pooled
+     * @param maxIdleTime max idle time when pruning is enabled
+     * @param timeUnit time unit of maxIdleTime
+     */
+    public PowerOfTwoBufferPool(final int highestPowerOfTwo, final long maxIdleTime, final TimeUnit timeUnit) {
         int powerOfTwo = 1;
         for (int i = 0; i <= highestPowerOfTwo; i++) {
-            final int size = powerOfTwo;
-            powerOfTwoToPoolMap.put(i, new ConcurrentPool<>(INFINITE_SIZE,
-                                                                         new ConcurrentPool.ItemFactory<ByteBuffer>() {
-                                                                             @Override
-                                                                             public ByteBuffer create() {
-                                                                                 return createNew(size);
-                                                                             }
-
-                                                                             @Override
-                                                                             public void close(final ByteBuffer byteBuffer) {
-                                                                             }
-
-                                                                             @Override
-                                                                             public Prune shouldPrune(final ByteBuffer byteBuffer) {
-                                                                                 return Prune.STOP;
-                                                                             }
-                                                                         }));
+            int size = powerOfTwo;
+            powerOfTwoToPoolMap.put(i, new ConcurrentPool<>(INFINITE_SIZE, new ItemFactory(size)));
             powerOfTwo = powerOfTwo << 1;
         }
+        maxIdleTimeNanos = timeUnit.toNanos(maxIdleTime);
+        pruner = Executors.newSingleThreadScheduledExecutor(new DaemonThreadFactory("BufferPoolPruner"));
+    }
+
+    /**
+     * Call this method at most once to enable a background thread that prunes idle buffers from the pool
+     */
+    public void enablePruning() {
```

Comment:
If we start returning `this` here, we could write `DEFAULT = new PowerOfTwoBufferPool().enablePruning()` and get rid of the `static` block.